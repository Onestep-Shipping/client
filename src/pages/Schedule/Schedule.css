.schedule-body-container {
  width: 100%;
  height: 85%;
  display: flex;
  flex-direction: column;
  padding: 50px 0px;
  align-items: center;
  background-color: white;
}

.schedule-result-container {
  width: 90%;
  border-top: solid #BF037A 1px;
  padding: 30px 0px;
}

.schedule-result-header-text {
  font-size: 18px;
  font-weight: bold;
}

.schedule-result-header-row {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 20px;
}

.schedule-result-row {
  width: 100%;
  height: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 10px;
  padding: 10px 0px;
  user-select: none;
  border-radius: 10px;
}

.schedule-result-row-selected {
  width: 100%;
  height: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 10px 0px;
  background-color: rgb(148, 223, 179);
  border-radius: 10px;
  user-select: none;
  margin-bottom: 10px;
}

.schedule-result-row-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.quote-dropdown {
  width: 50%;
  background-color: rgb(160, 238, 193);
  margin-bottom: 10px;
  border-bottom-right-radius: 10px;
  border-bottom-left-radius: 10px;
  padding: 20px;
  transition: all 0.3s ease-in;
  user-select: none;
}

.schedule-result-row:hover {
  background-color: rgb(230, 230, 230);
  border-radius: 10px;
}

.schedule-result-text {
  font-size: 16px;
}

.schedule-result-text-time {
  font-size: 15px;
  font-style: italic;
  color: rgb(77, 77, 77);
}

.col {
  width: 10%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.col2 {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.col3 {
  width: 30%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.result-button {
  padding: 10px 15px;
  background-color: #BF037A;
  border: none;
  outline: none;
  color: white;
  font-size: 20px;
  border-radius: 3px;
  cursor: pointer;
}

.result-button:hover {
  background-color: rgb(163, 0, 103);
}

.schedule-button-container {
  width: 15%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
}

.warning-text {
  color: #BF037A;
  font-size: 25px;
  font-weight: bolder;
}