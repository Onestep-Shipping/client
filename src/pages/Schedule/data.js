const DATA = [
  {
    id: 1,
    from: 'Vancouver, BC | CA',
    fromDate: '04/21/2020',
    trans: 0,
    ves: 'YM UPSURGENCE / 043W / PN2',
    to: 'Vung Tau | VN',
    toDate: '04/21/2020',
    time: '35',
  },
  {
    from: 'Vancouver, BC | CA',
    fromDate: '04/21/2020',
    trans: 0,
    ves: 'YM UPSURGENCE / 043W / PN2',
    to: 'Vung Tau | VN',
    toDate: '04/21/2020',
    time: '35',
  },
  {
    from: 'Vancouver, BC | CA',
    fromDate: '04/21/2020',
    trans: 0,
    ves: 'YM UPSURGENCE / 043W / PN2',
    to: 'Vung Tau | VN',
    toDate: '04/21/2020',
    time: '35',
  },
  {
    from: 'Vancouver, BC | CA',
    fromDate: '04/21/2020',
    trans: 0,
    ves: 'YM UPSURGENCE / 043W / PN2',
    to: 'Vung Tau | VN',
    toDate: '04/21/2020',
    time: '35',
  },
  {
    from: 'Vancouver, BC | CA',
    fromDate: '04/21/2020',
    trans: 0,
    ves: 'YM UPSURGENCE / 043W / PN2',
    to: 'Vung Tau | VN',
    toDate: '04/21/2020',
    time: '35',
  },
];

export default DATA;
