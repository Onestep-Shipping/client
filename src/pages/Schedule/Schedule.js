import React, {useState, useContext} from 'react';
import './Schedule.css';

import DATA from './data.js';
import Header from '../../components/Header/Header.js';
import ScheduleFormMin from '../../components/ScheduleForm/ScheduleFormMin.js';
import {AuthContext} from '../../firebase/Auth.js';

const Schedule = () => {
  const [resultHeaders] = useState(['Port of Loading', 'Transshipments',
    'Vessels / Services', 'Port of Discharge', 'Transit Time']);

  const [results] = useState(DATA);
  const [currentBookingIndex, setCurrentBookingIndex] = useState(0);

  const {currentUser} = useContext(AuthContext);

  return (
    <div class="homepage-container">
      <Header />
      <div class="schedule-body-container">
        <ScheduleFormMin />
        <div class="schedule-result-container">
          <div class="schedule-result-header-row">
            {
              resultHeaders.map((header) => {
                return (
                  <text class="schedult-result-header-text">{header}</text>
                );
              })
            }
          </div>
          {
            results.map((booking, index) => {
              return (
                <div
                  class={'schedule-result-row' +
                    (index === currentBookingIndex ? '-selected' : '')}
                  onClick={(e) =>setCurrentBookingIndex(index)}>
                  <div class="col">
                    <text class="schedult-result-text">{booking.from}</text>
                    <text class="schedult-result-text-time">{booking.fromDate}</text>
                  </div>
                  <div class="col">
                    <text class="schedult-result-text">{booking.trans}</text>
                  </div>
                  <div class="col">
                    <text class="schedult-result-text">{booking.ves}</text>
                  </div>
                  <div class="col">
                    <text class="schedult-result-text">{booking.to}</text>
                    <text class="schedult-result-text-time">{booking.toDate}</text>
                  </div>
                  <div class="col">
                    <text class="schedult-result-text">{booking.time}</text>
                  </div>
                </div>
              );
            })
          }
          {currentUser ?
            (<div class="schedule-button-container">
              <button class="result-button">Quote</button>
              <button class="result-button">Book</button>
            </div>) : (
              <a href="/auth" class="warning-text">
                You must log in to proceed.
              </a>
            )
          }
        </div>
      </div>
    </div>
  );
};

export default Schedule;
