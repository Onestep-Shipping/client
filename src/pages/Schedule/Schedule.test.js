import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import Schedule from './Schedule';
import { shallow } from 'enzyme';

let wrapper = shallow(
  <MemoryRouter>
    <Schedule />
  </MemoryRouter>
);

describe('Schedule', () => {
  it('should render the Schedule Component correctly', () => {   
    expect(Schedule).toMatchSnapshot();
  });
});