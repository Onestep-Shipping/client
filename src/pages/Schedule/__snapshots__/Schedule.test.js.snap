// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule should render the Schedule Component correctly 1`] = `[Function]`;
